commit db139efcfc7ec973ab1454117371b569f6711107
Author: John Levon <john.levon@joyent.com>
Date:   2019-03-14T16:30:51+00:00 (7 months ago)
    
    OS-7639 ctfconvert -i option is mis-handled
    OS-7659 mdb_v8 build is not CTF-complete

diff --git a/Makefile.defs b/Makefile.defs
index 92537db..75b445e 100644
--- a/Makefile.defs
+++ b/Makefile.defs
@@ -174,7 +174,7 @@ APPLY_PATCHES.64 = \
 	    $(GPATCH) -d .unpack64/$(VER) -p$(PATCHSTRIP) < "$$p" || exit 1; \
 	    done
 
-MAKE_CTF =		$(BASE)/../tools/make-ctf
+MAKE_CTF =		CTFCONVERTFLAGS=$(CTFCONVERTFLAGS) $(BASE)/../tools/make-ctf
 MAKE_CTF.32 =		$(MAKE_CTF) $(OBJ.32) $(CTF.32) $(VER)
 MAKE_CTF.64 =		$(MAKE_CTF) $(OBJ.64) $(CTF.64) $(VER)
 
diff --git a/mdb_v8/Makefile b/mdb_v8/Makefile
index 5c60cac..1aeb337 100644
--- a/mdb_v8/Makefile
+++ b/mdb_v8/Makefile
@@ -36,7 +36,8 @@ CTF_LIBS = $(VER)/build/ia32/mdb_v8.so \
 #
 # Required for CTF generation
 #
-CFLAGS += -g
+CFLAGS += -gdwarf-2
+CFLAGS_ARCH += -gdwarf-2
 
 #
 # To avoid "error: left shift of negative value"
@@ -51,7 +52,8 @@ CFLAGS += -Wno-shift-negative-value
 #
 all: $(VER)/GNUMakefile
 	cd $(VER) && \
-	    env - $(AUTOCONF_ENV) PATH=$(PATH) $(MAKE) $(OVERRIDES) release
+	    env - $(AUTOCONF_ENV) CFLAGS_ARCH=$(CFLAGS_ARCH) \
+	    PATH=$(PATH) $(MAKE) $(OVERRIDES) release
 
 install: all
 	rm -f $(DESTDIR)/usr/lib/mdb/proc/v8.so
diff --git a/openlldp/Makefile b/openlldp/Makefile
index e694c4a..20798da 100644
--- a/openlldp/Makefile
+++ b/openlldp/Makefile
@@ -18,7 +18,7 @@
 #
 # CDDL HEADER END
 #
-# Copyright (c) 2013, Joyent, Inc.  All rights reserved.
+# Copyright (c) 2019, Joyent, Inc.  All rights reserved.
 #
 
 VER =	openlldp-0.4alpha
@@ -41,6 +41,11 @@ CTF_PROGRAMS_USR_SBIN = \
 
 CTF_PROGRAMS = $(CTF_PROGRAMS_USR_SBIN:%=/usr/sbin/%)
 
+#
+# The upstream source has empty C files that we don't want to mess with.
+#
+CTFCONVERTFLAGS += -m
+
 all: all_autoconf
 
 install: all
diff --git a/tools/make-ctf b/tools/make-ctf
index 918fea6..64a9a54 100755
--- a/tools/make-ctf
+++ b/tools/make-ctf
@@ -63,7 +63,7 @@ for program in ${PROGRAMS}; do
 	program_path="${DESTDIR}${program}"
 	printf "  CTFCONV %s\n" "${program_path}" >&2
 	chmod u+w "${program_path}"
-	${CTFCONVERT} -l "${CONTAINER_VERSION}" \
+	${CTFCONVERT} ${CTFCONVERTFLAGS} -l "${CONTAINER_VERSION}" \
 	    -o "${program_path}" "${program_path}"
 	if [[ -z $NO_STRIP_SOURCE_DEBUG ]]; then
 		printf "  STRIP      %s\n" "${program_path}" >&2
